test/unit/app/controllers/preferences-controller-test.js

```diff
@@ -478,46 +478,16 @@ describe('preferences controller', function () {
-      } catch (e) {}
-      assert.equal(validateSpySymbol.threw(), true, 'options object with no symbol')
-      const validateSpyDecimals = sandbox.spy(preferencesController._validateERC20AssetParams)
-      try {
-        validateSpyDecimals({ rawAddress: '0xd26114cd6EE289AccF82350c8d8487fedB8A0C07', symbol: 'ABC' })
-      } catch (e) {}
-      assert.equal(validateSpyDecimals.threw(), true, 'options object with no decimals')
-      const validateSpyInvalidSymbol = sandbox.spy(preferencesController._validateERC20AssetParams)
-      try {
-        validateSpyInvalidSymbol({ rawAddress: '0xd26114cd6EE289AccF82350c8d8487fedB8A0C07', symbol: 'ABCDEFGHI', decimals: 0 })
-      } catch (e) {}
-      assert.equal(validateSpyInvalidSymbol.threw(), true, 'options object with invalid symbol')
-      const validateSpyInvalidDecimals1 = sandbox.spy(preferencesController._validateERC20AssetParams)
-      try {
-        validateSpyInvalidDecimals1({ rawAddress: '0xd26114cd6EE289AccF82350c8d8487fedB8A0C07', symbol: 'ABCDEFGHI', decimals: -1 })
-      } catch (e) {}
-      assert.equal(validateSpyInvalidDecimals1.threw(), true, 'options object with decimals less than zero')
-      const validateSpyInvalidDecimals2 = sandbox.spy(preferencesController._validateERC20AssetParams)
-      try {
-        validateSpyInvalidDecimals2({ rawAddress: '0xd26114cd6EE289AccF82350c8d8487fedB8A0C07', symbol: 'ABCDEFGHI', decimals: 38 })
-      } catch (e) {}
-      assert.equal(validateSpyInvalidDecimals2.threw(), true, 'options object with decimals more than 36')
-      const validateSpyInvalidAddress = sandbox.spy(preferencesController._validateERC20AssetParams)
-      try {
-        validateSpyInvalidAddress({ rawAddress: '0x123', symbol: 'ABC', decimals: 0 })
-      } catch (e) {}
-      assert.equal(validateSpyInvalidAddress.threw(), true, 'options object with address invalid')
+      const validate = preferencesController._validateERC20AssetParams
+
+      assert.doesNotThrow(() => validate({ rawAddress: '0xd26114cd6EE289AccF82350c8d8487fedB8A0C07', symbol: 'ABC', decimals: 0 }))
+      assert.throws(() => validate({ symbol: 'ABC', decimals: 0 }), 'missing address should fail')
+      assert.throws(() => validate({ rawAddress: '0xd26114cd6EE289AccF82350c8d8487fedB8A0C07', decimals: 0 }), 'missing symbol should fail')
+      assert.throws(() => validate({ rawAddress: '0xd26114cd6EE289AccF82350c8d8487fedB8A0C07', symbol: 'ABC' }), 'missing decimals should fail')
+      assert.throws(() => validate({ rawAddress: '0xd26114cd6EE289AccF82350c8d8487fedB8A0C07', symbol: 'ABCDEFGHI', decimals: 0 }), 'invalid symbol should fail')
+      assert.throws(() => validate({ rawAddress: '0xd26114cd6EE289AccF82350c8d8487fedB8A0C07', symbol: 'ABCDEFGHI', decimals: -1 }), 'decimals < 0 should fail')
+      assert.throws(() => validate({ rawAddress: '0xd26114cd6EE289AccF82350c8d8487fedB8A0C07', symbol: 'ABCDEFGHI', decimals: 38 }), 'decimals > 36 should fail')
```

Comment:
Welp; these two tests have always been broken. They'd fail because of the symbol, not because of the decimals.
```suggestion
      assert.throws(() => validate({ rawAddress: '0xd26114cd6EE289AccF82350c8d8487fedB8A0C07', symbol: 'ABC', decimals: -1 }), 'decimals < 0 should fail')
      assert.throws(() => validate({ rawAddress: '0xd26114cd6EE289AccF82350c8d8487fedB8A0C07', symbol: 'ABC', decimals: 38 }), 'decimals > 36 should fail')
```

Out-of-scope for this PR, just mentioning it.

Reply:
I'll update these separately 👍